ample/ensembler/truncation_util.py

```diff
@@ -32,204 +39,98 @@ def calculate_residues_focussed(var_by_res):
     For > 80 residues, split < 40 into 10 4-residue chunks, and split the interval 40 -> end into
     10 even chunks.
     """
-    
     length = len(var_by_res)
     if length <= 80:
         # Just split evenly into 20 chunks
-        return calculate_residues_percent(var_by_res, 5)
-
-    # Get list of residue indices sorted by variance - from least variable to most
+        return calculate_residues_by_percent(var_by_res, percent_truncation=5)
     var_by_res.sort(key=lambda x: x.variance, reverse=False)
-     
     # Split a 40 - length interval into 10 even chunks.
     llen = 40
     lower_start = _split_sequence(llen, 10)
-    
     # Split remaining interval into 10 even chunks. We need to add the start sequence as we have
     # removed llen residues
     ulen = length - llen
     upper_start = [ i + llen for i in _split_sequence(ulen, 10) ]
     start_indexes = upper_start + lower_start 
-    
     # Calculate the percentages for each of these start points
     percentages = [ int(round(float(start + 1) / float(length) * 100)) for start in start_indexes ]
     # print "percentages ", percentages
     truncation_levels = percentages
-
-    # print "var_by_res ",var_by_res
     idxs_all = [ x.idx for x in var_by_res ]
     resseq_all = [ x.resSeq for x in var_by_res ]
     variances = [ x.variance for x in var_by_res ]
-
     truncation_residue_idxs = [ sorted(idxs_all[:i + 1]) for i in start_indexes ]
-    # print "truncation_residue_idxs ",truncation_residue_idxs
     truncation_residues = [ sorted(resseq_all[:i + 1]) for i in start_indexes ]
-    # print "truncation_residues ",truncation_residues
-    
     # We take the variance of the most variable residue
     truncation_variances = [ variances[i] for i in start_indexes ] 
-    # print "truncation_variances ",truncation_variances
     
     return truncation_levels, truncation_variances, truncation_residues, truncation_residue_idxs
 
 
-def calculate_residues_percent(var_by_res, percent_interval):
-    """Calculate the list of residues to keep if we are keeping self.percent residues under
-    each truncation bin. The threshold is just the threshold of the most variable residue"""
-    
-    MIN_CHUNK = 3  # We need at least 3 residues for theseus to work
-    length = len(var_by_res)
-    start_idxs = _split_sequence(length, percent_interval, min_chunk=MIN_CHUNK)
+def _calculate_start_indexes_from_fixed_percentages(percent_fixed_intervals, length, all_indexes):
+    "Calculate where in the list of residues each percentage bin starts"
+    def find_closest_index(nresidues_ideal, all_indexes):
+        min_dist = 99999 
+        start = None
+        for idx in all_indexes:
+            rdist = abs(nresidues_ideal - (idx + 1))
+            if  rdist <= min_dist and idx + 1 >= MIN_CHUNK:
+                min_dist = rdist
+                start = idx
+        assert start is not None
```

Comment:
Better to raise an Exception than the assert here. It won't make any sense when receiving log files containing this error.

Reply:
It's just a failsafe that should never be triggered